deltadb/src/catalog_export.c

```diff
@@ -160,3 +159,39 @@ void catalog_export_html_footer( struct link *l, struct jx_table *h, time_t stop
 {
 	link_printf(l,stoptime,"</table>\n");
 }
+
+void catalog_export_html_datetime_picker( struct link *l, time_t stoptime, time_t current) {
+	struct tm *t1 = gmtime(&current);
+	struct tm *t2 = gmtime(&current);
+	struct tm *t3 = gmtime(&current);
+
+	int year = t1->tm_year + 1900;
+	int month = t1->tm_mon + 1;
+	int day = t1->tm_mday;
+
+	t2->tm_mday--;
```

Comment:
What happens if tm_mday goes negative here?

Reply:
mktime handles that for us :)

https://www.cplusplus.com/reference/ctime/mktime/

> The values of the members tm_wday and tm_yday of timeptr are ignored, and the values of the other members are interpreted even if out of their valid ranges (see [struct tm](https://www.cplusplus.com/tm)). For example, tm_mday may contain values above 31, which are interpreted accordingly as the days that follow the last day of the selected month.
> 
> A call to this function automatically adjusts the values of the members of timeptr if they are off-range or -in the case of tm_wday and tm_yday- if they have values that do not match the date described by the other members.